src/Dev/Deprecation.php

```diff
@@ -163,8 +174,53 @@ protected static function get_called_method_from_trace($backtrace, $level = 1)
         if ($level == 4 && ($backtrace[$newLevel]['class'] ?? '') === InjectionCreator::class) {
             $newLevel = $newLevel + 4;
         }
+        // handle noticeWithNoReplacment()
+        foreach ($backtrace as $trace) {
+            if (($trace['class'] ?? '') === Deprecation::class
+                && ($trace['function'] ?? '') === 'noticeWithNoReplacment'
+            ) {
+                $newLevel = $newLevel + 1;
+                break;
+            }
+        }
         $called = $backtrace[$newLevel] ?? [];
-        return ($called['class'] ?? '') . ($called['type'] ?? '') . ($called['function'] ?? '');
+        return $called;
+    }
+
+    private static function calledFromSupportedCode(array $backtrace): bool
```

Comment:
```suggestion
    private static function isCalledFromSupportedCode(array $backtrace): bool
```
This matches the style used for `isSupportedVedorFile()` below.
Though since it's private and the name explains what it does without this, I won't block merging if you don't change this.

Reply:
Actually if feels like this should just be blended together with `isSupportedVendorFile()`:

```php
private static function isCalledFromSupportedCode(array $backtrace): bool
{
    $called = Deprecation::get_called_from_trace($backtrace, 1);
    if (!isset($called['file'])) {
        return false;
    }
    $file = $called['file'];
    // and then all of the code from isSupportedVendorFile() goes here
}
```

Reply:
Updated